### PR TITLE
fix: missing redeploy of chart from zero instances

### LINKED ISSUE #2161: Application does not appear after scaling to 0 instances and back to 1+
**Describe the bug**
After scaling an existing app to 0 instances and then scaling back to a positive amount of instances the app won't re-appear.

Visible in `epinio:v1.7.1`

**To Reproduce**
```
$ epinio app update sample --instances 0
$ epinio app list

🚢  Listing applications
Namespace: workspace

✔️  Epinio Applications:
|  NAME  |            CREATED            | STATUS | ROUTES | CONFIGURATIONS | STATUS DETAILS |
|--------|-------------------------------|--------|--------|----------------|----------------|
| sample | 2023-03-30 11:18:49 +0000 UTC | n/a    | n/a    |                |                |
$ epinio app update sample --instances 3
$ epinio app list

🚢  Listing applications
Namespace: workspace

✔️  Epinio Applications:
|  NAME  |            CREATED            | STATUS | ROUTES | CONFIGURATIONS | STATUS DETAILS |
|--------|-------------------------------|--------|--------|----------------|----------------|
| sample | 2023-03-30 11:18:49 +0000 UTC | n/a    | n/a    |                |                |
$ kubectl get pods -n workspace
No resources found in workspace namespace.
```

**Expected behavior**
the workload should re-appear with given amount of instances

### PR DESCRIPTION
**Pull request by andreas-kupries**
fix #2161

chore: additional tracing of update ops

### CODE REVIEW

**Comment by mmartin24** [comment]
I checked this issue and I can scale down to 0:
```
➜  epinio git:(2161-scale-up-from-nothing) epinio app list

🚢  Listing applications
Namespace: workspace

✔️  Epinio Applications:
|   NAME    |            CREATED             | STATUS | ROUTES | CONFIGURATIONS | STATUS DETAILS |
|-----------|--------------------------------|--------|--------|----------------|----------------|
| testapp-1 | 2023-04-04 09:48:50 +0200 CEST | 0/0    | n/a    |                |                |

```
However when I scale up to 4 I get status `n/a` and `deployment failed`:
```
  epinio git:(2161-scale-up-from-nothing) epinio app update testapp-1 --instances 4

🚢  Update application
Namespace: workspace
Application: testapp-1

✔️  Successfully updated application
➜  epinio git:(2161-scale-up-from-nothing) epinio app list                          

🚢  Listing applications
Namespace: workspace

✔️  Epinio Applications:
|   NAME    |            CREATED             | STATUS | ROUTES | CONFIGURATIONS |  STATUS DETAILS   |
|-----------|--------------------------------|--------|--------|----------------|-------------------|
| testapp-1 | 2023-04-04 09:48:50 +0200 CEST | n/a    | n/a    |                | deployment failed |

```
@andreas-kupries does it occur to you as well, or maybe I am doing something wrong?

**Comment by mmartin24** [comment]
I retested to check if perhaps the [error](https://github.com/epinio/epinio/pull/2174#issuecomment-1495514785) above-mentioned occured perhaps for scaling down too quickly not ensuring all pods were healthy. It turned out I could not see the 0/0 status.
 
 ```
✔️  Epinio Applications:
|   NAME    |            CREATED             | STATUS |                  ROUTES                   | CONFIGURATIONS |    STATUS DETAILS     |
|-----------|--------------------------------|--------|-------------------------------------------|----------------|-----------------------|
| testapp-1 | 2023-04-04 10:51:41 +0200 CEST | 1/1    | testapp-1.192.168.16.3.omg.howdoi.website |                | metrics not available |
➜  epinio git:(2161-scale-up-from-nothing) epinio app update testapp-1 --instances 0    

🚢  Update application
Namespace: workspace
Application: testapp-1

✔️  Successfully updated application
➜  epinio git:(2161-scale-up-from-nothing) epinio app list                          

🚢  Listing applications
Namespace: workspace

✔️  Epinio Applications:
|   NAME    |            CREATED             | STATUS | ROUTES | CONFIGURATIONS | STATUS DETAILS |
|-----------|--------------------------------|--------|--------|----------------|----------------|
| testapp-1 | 2023-04-04 10:51:41 +0200 CEST | n/a    | n/a    |                |                |

```
At first glance I see everything running:
```
➜  epinio git:(2161-scale-up-from-nothing) k get svc -A 
NAMESPACE      NAME                                                  TYPE           CLUSTER-IP      EXTERNAL-IP    PORT(S)                      AGE
default        kubernetes                                            ClusterIP      10.43.0.1       <none>         443/TCP                      17m
kube-system    kube-dns                                              ClusterIP      10.43.0.10      <none>         53/UDP,53/TCP,9153/TCP       17m
kube-system    metrics-server                                        ClusterIP      10.43.145.109   <none>         443/TCP                      17m
kube-system    traefik                                               LoadBalancer   10.43.154.142   192.168.16.3   80:30373/TCP,443:32239/TCP   16m
cert-manager   cert-manager                                          ClusterIP      10.43.118.96    <none>         9402/TCP                     16m
cert-manager   cert-manager-webhook                                  ClusterIP      10.43.127.80    <none>         443/TCP                      16m
epinio         minio-svc                                             ClusterIP      None            <none>         9000/TCP                     15m
epinio         epinio-ui                                             ClusterIP      10.43.214.128   <none>         80/TCP                       15m
epinio         kubed                                                 ClusterIP      10.43.202.246   <none>         443/TCP                      15m
epinio         minio                                                 ClusterIP      10.43.43.138    <none>         9000/TCP                     15m
epinio         minio-console                                         ClusterIP      10.43.77.115    <none>         9001/TCP                     15m
epinio         registry                                              ClusterIP      10.43.172.30    <none>         5000/TCP                     15m
epinio         dex                                                   ClusterIP      10.43.162.152   <none>         5556/TCP,5558/TCP            15m
epinio         epinio-server                                         ClusterIP      10.43.144.124   <none>         80/TCP                       15m
epinio         registry-node                                         NodePort       10.43.85.212    <none>         30500:30500/TCP              15m
workspace      rtestapp-1-f2da56eca818fa5e7f16dd40a65472943ca16412   ClusterIP      10.43.59.61     <none>         8080/TCP                     6m10s
```
 To set up the environment I deployed k3d using:
 
```
make acceptance-cluster-delete && make acceptance-cluster-setup && k3d kubeconfig merge -ad && kubectl config use-context k3d-epinio-acceptance && make install-cert-manager && make prepare_environment_k3d && make && make patch-epinio-deployment
```
I am not sure if I am missing anything but I am not able to see it working

**Comment by andreas-kupries** [comment]
The PR did not contain the work on the status messages.
The recent force-push rebased the PR to main where this fix is available.

File: acceptance/apps_test.go [hunk 1]
```diff
@@ -1643,6 +1643,22 @@ userConfig:
 					),
 				)
 			})
+
+			It("scales up to a running workload", func() {
+				out, err := env.Epinio("", "app", "update", app, "-i", "3")
```
Comment (mmartin24) [inline acceptance/apps_test.go:1648]: Just a question. Not sure about what `-i` is here. Does this test check from 0 to 3?

Comment (mmartin24) [reply acceptance/apps_test.go:1648]: ok. nvm. It seems so. I  saw this later
```
BeforeEach(func() {
				app = catalog.NewAppName()
				By("make zero-instance app: " + app)
				env.MakeApp(app, 0, false)
```

File: internal/api/v1/application/update.go [hunk 2]
```diff
@@ -59,6 +60,8 @@ func (hc Controller) Update(c *gin.Context) apierror.APIErrors { // nolint:gocyc
 		return apierror.NewBadRequestError(err.Error())
 	}
 
+	log.Info("updating app", "namespace", namespace, "app", appName, "request", updateRequest)
```
Comment (mmartin24) [inline internal/api/v1/application/update.go:63]: the purpose of  `log.Info` is to print out a log (like `echo`in Bash or `console.log` in JS)?

Comment (andreas-kupries) [reply internal/api/v1/application/update.go:63]: Yes. I added quite a bit more to better see what was happening. The output is seen in the kube log of the epinio server pod. --- `k logs -c epinio-server -f :epinio-server`

Comment (mmartin24) [reply internal/api/v1/application/update.go:63]: nvm. It seems so. I knew about `fmt` and wondered about the differences. lt seems log is primarily used for logging messages to a log file or standard error stream, while fmt is used for general-purpose formatting and printing.

File: internal/api/v1/application/update.go [hunk 3]
```diff
@@ -142,14 +147,19 @@ func (hc Controller) Update(c *gin.Context) apierror.APIErrors { // nolint:gocyc
...
 
+		log.Info("updating app", "app chart patch", patch)
+
 		_, err = client.Namespace(app.Meta.Namespace).Patch(ctx, app.Meta.Name, types.JSONPatchType, []byte(patch), metav1.PatchOptions{})
 		if err != nil {
 			return apierror.InternalError(err)
 		}
 	}
 
+	var desired int32
```
[the remark this answers is not in the document]
Comment (mmartin24) [reply internal/api/v1/application/update.go:158]: Solved offline.
The reason is to match previously set `Instances` here: https://github.com/epinio/epinio/blob/main/pkg/api/core/v1/models/models.go#L178

**Review by mmartin24 — APPROVED** [review]
LGTM after reviewing / solving questions with Andreas.

**Comment by thehejik** [comment]
Verified fixed in `v1.8.0-rc1`
```
epinio push -n testapp -p epinio/assets/sample-app/

**Pull request merged**